src/main/groovy/com/ullink/testtools/elastic/TestExportTask.groovy

```diff
@@ -114,10 +117,19 @@ class TestExportTask extends Exec {
                 String index = indexPrefix + timetamp.format(DateTimeFormatter.ofPattern(indexTimestampPattern))
                 index = index.replace('.', '-')
                 String typeFinal
-                if (type instanceof String)
-                    typeFinal = type
-                if (type instanceof Closure)
-                    typeFinal = type.call()
+                switch (type.getClass()) {
+                    case GString:
+                        typeFinal = type.toString()
+                        break
+                    case String:
+                        typeFinal = type
+                        break
+                    case Closure:
```

Comment:
Closure being an abstract class, getClass == Closure can never be true, can it ?

Reply:
my bad I should have checked on the instance itself, not the class so groovy will check if it is an instance of

Reply:
I've amended the change